File: src/lessons/welcome/variables/RunFourEntity.java
package lessons.welcome.variables;

public class RunFourEntity extends plm.universe.bugglequest.SimpleBuggle {
	@Override
	public void forward(int i)  { 
		throw new RuntimeException(getGame().i18n.tr("Sorry Dave, I cannot let you use forward with an argument in this exercise. Use a loop instead."));
	}
	@Override
	public void backward(int i) {
		throw new RuntimeException(getGame().i18n.tr("Sorry Dave, I cannot let you use backward with an argument in this exercise. Use a loop instead."));
	}


	@Override
	/* BEGIN TEMPLATE */
	public void run() { 
		/* BEGIN SOLUTION */
		int cpt = 0;
		while (cpt != 4) {
			forward();
			if (isOverBaggle())
				cpt++;
		}
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}

File: src/lessons/welcome/variables/RunFourEntity.py
def forward(i=1):
    if i>1:
        errorMsg("Sorry Dave, I cannot let you use forward(i>1) in this exercise")
    entity.forward()
def backward(i=1):
    if i>1:
        errorMsg("Sorry Dave, I cannot let you use backward(i>1) in this exercise")
    entity.backward()

# BEGIN SOLUTION
cpt = 0
while cpt != 4:
	forward()
	if isOverBaggle():
		cpt += 1
# END SOLUTION 
